src/compiler/checker.ts

```diff
@@ -25535,12 +25535,40 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         errorOrSuggestion(noImplicitAny, declaration, diagnostic, declarationNameToString(getNameOfDeclaration(declaration)), typeAsString);
     }
 
+    function reportErrorsFromWideningWithContextualSignature(declaration: FunctionLikeDeclaration, wideningKind: WideningKind) {

```

Comment:
Rename to `shouldReportErrorsFromWideningWithContextualSignature` given this returns a boolean, it doesn't actually report?